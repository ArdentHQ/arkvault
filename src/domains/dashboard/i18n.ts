export const translations = {
	DASHBOARD_PAGE: {
		TITLE: "Dashboard",
	},
	FILTER_WALLETS: {
		CRYPTOASSET: {
			DESCRIPTION: "Select the networks to show.",
			TITLE: "Filter Networks",
		},
		WALLETS: {
			DESCRIPTION: "Select which wallets to view.",
			TITLE: "Your Wallets",
		},
	},

	LATEST_TRANSACTIONS: {
		EMPTY_MESSAGE:
			"You don't have any transactions yet. Once transactions have been made they will show up in your history.",
		NO_RESULTS: "No <bold>{{type}}</bold> transactions could be found.",
		TITLE: "Latest Transactions",
	},

	PORTFOLIO_BREAKDOWN: {
		FILTERED: "Please enable at least one public network to display your portfolio report.",
	},

	PORTFOLIO_BREAKDOWN_DETAILS: {
		TITLE: "Portfolio Breakdown",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		EMPTY_MESSAGE: "Your portfolio is currently empty. <1>Create</1> or <3>Import</3> a wallet to get started.",
		EMPTY_MESSAGE_FILTERED: "You don't have any wallets matching the selected networks.",
		EMPTY_MESSAGE_TYPE: "You don't have any <bold>{{type}}</bold> wallets.",
		EMPTY_MESSAGE_TYPE_FILTERED: "You don't have any <bold>{{type}}</bold> wallets matching the selected networks.",
		IMPORT: "Import",
		IMPORT_LEDGER: "Import Ledger",
	},
};
