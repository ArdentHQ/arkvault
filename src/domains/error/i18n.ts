export const translations = {
	APPLICATION: {
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		HELP_TEXT: "Please provide the error below when contacting support.",
		RELOAD: "Reload",
		TITLE: "Something went wrong",
	},
	OFFLINE: {
		DESCRIPTION: "It looks like there may be a problem connecting to the Internet. Check your connection.",
		TITLE: "Connectivity Issue",
	},
};
