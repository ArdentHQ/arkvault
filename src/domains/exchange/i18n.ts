export const translations = {
	EMPTY_MESSAGE: "You haven't made any swaps yet. Once you initiate a transaction they will show up here.",

	ERROR: {
		GENERIC: "Your order could not be placed",
		INVALID_ADDRESS: "The provided address is not a valid {{ticker}} address",
		INVALID_REFUND_ADDRESS: "The provided refund address is not a valid {{ticker}} address",
	},

	EXCHANGE_FORM: {
		AMOUNT_RECEIVED: "Amount Received",
		AMOUNT_SENT: "Amount Sent",
		CURRENCY_INPUT: "{{currency}} Input",
		CURRENCY_OUTPUT: "{{currency}} Output",
		ESTIMATED_ARRIVAL: "Estimated Arrival",
		ESTIMATED_RATE: "Estimated Rate",
		ESTIMATED_TIME: "â‰ˆ {{estimatedTime}} minutes",
		EXCHANGE_ADDRESS: "Exchange {{currency}} Address",
		FROM_CURRENCY: "From Currency",
		INPUT_TRANSACTION_ID: "Input Transaction ID",
		OUTPUT_TRANSACTION_ID: "Output Transaction ID",
		PAYIN_AMOUNT: "Payin Amount",
		PAYOUT_AMOUNT: "Payout Amount",
		RECIPIENT_PLACEHOLDER: "Enter the payout address",
		RECIPIENT_WALLET: "Recipient Wallet",
		REFUND_PLACEHOLDER: "Enter the refund address",
		REFUND_WALLET: "Refund Wallet",
		SUPPORT_INFO:
			"If you have any question about your exchange, please contact {{exchange}} support team via email at <linkEmail>{{email}}</linkEmail>",
		TERMS: "I've read and agree to the {{exchange}} <linkTerms>{{terms}}</linkTerms> and <linkPrivacyPolicy>{{privacy}}</linkPrivacyPolicy>",
		TO_CURRENCY: "To Currency",
		TRANSACTION_COMPLETE: "Transaction is Complete!",
		YOUR_ADDRESS: "Your {{currency}} Address",
		YOU_GET: "You Get",
		YOU_SEND: "You Send",
	},

	EXCHANGE_NAME: "Exchange Name",
	EXPECTED_AMOUNT_HINT: "The final payout amount might be different",

	MODAL_DELETE_EXCHANGE_TRANSACTION: {
		DESCRIPTION:
			"Are you sure you want to remove this swap from the transaction history? This action cannot be undone.",
		TITLE: "Remove Transaction",
	},

	NAVIGATION: {
		EXCHANGES: "Exchanges",
		TRANSACTIONS: "Transactions",
	},

	PAGE_EXCHANGES: {
		ADD_EXCHANGE: "Add Exchange",
		DELETE_CONFIRMATION: "The exchange transaction <bold>{{orderId}}</bold> has been deleted",
		EMPTY_MESSAGE: "There are currently no exchange providers available, please try again at a later time.",
		SUBTITLE: "Choose one of the exchange providers below to swap directly from within the wallet.",
		TITLE: "Exchange",
	},

	REFUND_WALLET: {
		ADD: "Add Refund Wallet",
		REMOVE: "Remove Refund Wallet",
	},

	SELECT_EXCHANGE_MESSAGE: "Select one of the exchanges you have installed",

	STATUS: {
		CONFIRMING: "Confirming",
		EXCHANGING: "Exchanging",
		EXPIRED: "Expired",
		FAILED: "Failed",
		FINISHED: "Finished",
		NEW: "New",
		REFUNDED: "Refunded",
		SENDING: "Sending",
		VERIFYING: "Verifying",
		WAITING: "Waiting",
	},

	TO_ADDRESS: "To Address",

	TRANSACTION_ID: "Transaction ID",
	TRANSACTION_STATUS: {
		AWAITING_DEPOSIT: "Awaiting Deposit",
		EXCHANGING: "Exchanging",
		EXPIRED: "This transaction has expired.",
		FAILED: "This transaction has failed.",
		REFUNDED: "This transaction has been refunded.",
		SENDING: "Sending",
		VERIFYING: "This transaction is being verified by the provider.",
	},

	VALIDATION: {
		INVALID_ADDRESS: "Not a valid {{ticker}} address",
		MIN_AMOUNT: "The minimal exchange amount is {{amount}} {{ticker}}",
		PAIR_NOT_AVAILABLE: "The pair <bold>{{from}}</bold> / <bold>{{to}}</bold> is not available.",
	},

	YOUR_EXCHANGE_LIST: "Your Exchange List",
};
