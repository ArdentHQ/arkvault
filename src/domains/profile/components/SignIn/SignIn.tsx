import { Contracts } from "@ardenthq/sdk-profiles";
import React, { useCallback, useEffect, useState } from "react";
import { useForm } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { Button } from "@/app/components/Button";
import { Divider } from "@/app/components/Divider";
import { Form, FormButtons, FormField, FormLabel } from "@/app/components/Form";
import { InputPassword } from "@/app/components/Input";
import { Modal } from "@/app/components/Modal";
import { ProfileAvatar } from "@/domains/profile/components/ProfileAvatar";

interface SignInProperties {
	isOpen: boolean;
	profile: Contracts.IProfile;
	onCancel?: () => void;
	onClose?: () => void;
	onSuccess: (password: string) => void;
}

const MAX_ATTEMPTS = 3;
const TIMEOUT = 60;

export const SignIn: React.VFC<SignInProperties> = ({ isOpen, profile, onCancel, onClose, onSuccess }) => {
	const { t } = useTranslation();

	const methods = useForm({ mode: "onChange" });
	const { errors, formState, register, setError } = methods;

	const [count, setCount] = useState(0);
	const [remainingTime, setRemainingTime] = useState(0);

	const hasReachedLimit = useCallback(() => count >= MAX_ATTEMPTS, [count]);

	useEffect(() => {
		if (hasReachedLimit()) {
			setRemainingTime(TIMEOUT);
		}

		if (count) {
			const timer = setInterval(() => {
				setCount(count - 1);
			}, TIMEOUT * 1000);

			return () => clearInterval(timer);
		}
	}, [count, hasReachedLimit]);

	useEffect(() => {
		if (remainingTime) {
			setError("password", {
				message: t("PROFILE.MODAL_SIGN_IN.MAX_ATTEMPTS_ERROR", { remainingTime }),
				type: "maxAttempts",
			});

			const timer = setInterval(() => {
				setRemainingTime(remainingTime - 1);
			}, 1000);

			return () => clearInterval(timer);
		} else if (errors.password) {
			setError("password", {
				message: t("COMMON.VALIDATION.FIELD_INVALID", {
					field: t("COMMON.PASSWORD"),
				}),
				type: "invalid",
			});
		}
	}, [errors, remainingTime, setError, t]);

	const handleSubmit = ({ password }) => {
		if (!profile.auth().verifyPassword(password)) {
			setCount(count + 1);

			setError("password", {
				message: t("COMMON.VALIDATION.FIELD_INVALID", {
					field: t("COMMON.PASSWORD"),
				}),
				type: "invalid",
			});
			return;
		}

		onSuccess(password);
	};

	if (!isOpen) {
		return <></>;
	}

	return (
		<Modal
			title={t("PROFILE.MODAL_SIGN_IN.TITLE")}
			description={t("PROFILE.MODAL_SIGN_IN.DESCRIPTION")}
			size="md"
			isOpen={isOpen}
			onClose={onClose}
		>
			<div className="mt-8">
				<div className="flex items-center space-x-3">
					<ProfileAvatar profile={profile} />

					<div className="flex flex-col">
						<p className="text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700">
							{t("COMMON.NAME")}
						</p>
						<p className="font-semibold text-theme-text" data-testid="profile-card__user--name">
							<span>{profile.name()}</span>
						</p>
					</div>
				</div>
			</div>

			<Divider dashed={true} />

			<Form context={methods} onSubmit={handleSubmit} className="mt-8">
				<FormField name="password">
					<FormLabel label={t("SETTINGS.GENERAL.PERSONAL.PASSWORD")} />
					<InputPassword
						ref={register({
							required: t("COMMON.VALIDATION.FIELD_REQUIRED", {
								field: t("COMMON.PASSWORD"),
							}).toString(),
						})}
						disabled={hasReachedLimit()}
						data-testid="SignIn__input--password"
					/>
				</FormField>

				<FormButtons>
					<Button data-testid="SignIn__cancel-button" variant="secondary" onClick={onCancel}>
						{t("COMMON.CANCEL")}
					</Button>

					<Button data-testid="SignIn__submit-button" type="submit" disabled={!formState.isValid}>
						{t("COMMON.SIGN_IN")}
					</Button>
				</FormButtons>
			</Form>
		</Modal>
	);
};
