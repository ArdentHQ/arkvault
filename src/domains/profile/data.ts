// @TODO: could expose a method in profiles SDK to get default setting values
export const DEFAULT_MARKET_PROVIDER = "cryptocompare";
