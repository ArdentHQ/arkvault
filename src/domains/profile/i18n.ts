export const translations = {
	IMPORT: {
		FORM_STEP: {
			DESCRIPTION: "Your profile addresses have been imported.",
		},
		PASSWORD_DESCRIPTION:
			"The profile you are importing contains a password that has been set for security purposes. Enter the password to confirm.",
		PASSWORD_TITLE: "Profile Password",
		PROCESSING_IMPORT_STEP: {
			DESCRIPTION: "The file with name {{name}} is being imported.",
			ERROR: "An error occurred while importing the profile, try importing again. Select Retry to re-import the profile. If the error occurs again, you can contact our support team.",
		},
		SELECT_FILE_STEP: {
			BROWSE_FILES: "Browse Files",
			CLICK_HERE: "Click here.",
			DEPRECATION_WARNING:
				"Please note that importing a Profile from Desktop Wallet v2 in .json format is supported temporarily and will soon be disabled. We recommend that you re-save your Profile in the new .wwe format.",
			DESCRIPTION:
				"Select a {{fileFormat}} file with your Profile and related properties to start the import process.",
			DRAG_AND_DROP: "Drag & Drop or",
			ERRORS: {
				NOT_SUPPORTED:
					"This file type is not supported.\nPlease retry with a <strong>{{fileFormat}}</strong> file.",
				TOO_MANY: "You tried to import <strong>{{fileCount}}</strong> files.\nPlease retry with one file only.",
			},
			LEGACY_IMPORT: "Importing from Desktop Wallet v2?",
			SUPPORTED_FORMAT: "Supported format is {{fileFormat}}",
			UPLOAD_TITLE: "Click here to upload",
		},
		TITLE: "Import Profile",
	},
	MODAL_BETA_NOTICE: {
		DESCRIPTION:
			"Payvo is experimental and shall not create any obligation for ARK Ecosystem SCIC to continue to develop, productize, support, repair, offer for sale, or in any other way continue to provide or develop Software either to Licensee or to any other party.\nThe Software is provided «AS IS» without any express or implied warranty of any kind including warranties of merchantability or fitness for any particular purpose. In no event shall ARK Ecosystem SCIC or its subsidiaries, partners, or related entities be liable for any damages whatsoever (including, without limitation, damages for loss of profits or funds, business interruption, loss of information arising out of the use or inability to use the software).",
		TITLE: "Payvo Beta Testing",
	},
	MODAL_DELETE_PROFILE: {
		DESCRIPTION: "Do you really want to delete this profile? Once deleted, you will not be able to restore it.",
		TITLE: "Delete Profile",
	},
	MODAL_HISTORY: {
		TITLE: "History",
		TYPES: {
			REGISTRATION: "Registration",
			RESIGN: "Resign",
			UPDATE: "Update",
		},
	},
	MODAL_RESET_PROFILE: {
		DESCRIPTION:
			"By performing this action, all of your Profile settings will be restored to default. This won't delete your wallets. This action cannot be undone. Do you want to reset your settings?",
		SUCCESS: "Your profile has been reset successfully",
		TITLE: "Reset Profile Settings",
	},
	MODAL_SELECT_ADDRESS: {
		DESCRIPTION: "Locate and select the address for receiving funds.",
		SEARCH_PLACEHOLDER: "Enter name or address",
		TITLE: "Select Address",
	},
	MODAL_SELECT_SENDER: {
		DESCRIPTION: "Find and select the sender from your wallets",
		SEARCH_PLACEHOLDER: "Enter name or address",
		TITLE: "Select Sender",
	},
	MODAL_SIGN_IN: {
		DESCRIPTION: "Sign in to access your Profile.",
		MAX_ATTEMPTS_ERROR:
			"Maximum sign in attempts reached - please wait {{remainingTime}} seconds before trying again.",
		TITLE: "Sign In",
	},
	MODAL_WELCOME: {
		DONT_SHOW_CHECKBOX_LABEL: "Don't show this quick tour again.",
		STEP_1: {
			DESCRIPTION:
				"Start a quick tour to learn how to get the most out of the Payvo Wallet, or skip it to go directly to your profile.",
			TITLE: "Welcome to Payvo Wallet",
		},
		STEP_2: {
			DESCRIPTION:
				"Create or import existing wallets from a broad range of assets and secure them using industry-standard encryption.",
			TITLE: "Manage Your Assets",
		},
		STEP_3: {
			DESCRIPTION:
				"Personalize the wallet according to your preference - change currencies, themes, colors and more.",
			TITLE: "Fully Customizable",
		},
		STEP_4: {
			DESCRIPTION: "With Payvo, you’re in control. Swap assets effortlessly without ever leaving your wallet.",
			TITLE: "Exchange Assets On The Fly",
		},
		STEP_5: {
			DESCRIPTION: "Customize or search your news feed to view the latest updates on your favorite cryptoassets.",
			TITLE: "Stay Up To Date",
		},
	},
	PAGE_CREATE_PROFILE: {
		DESCRIPTION: "Create a new Profile below.",
		DISCLAIMER:
			"I accept the <linkTerms>Terms of Service</linkTerms> and <linkPrivacyPolicy>Privacy Policy</linkPrivacyPolicy>",
		TITLE: "Create Profile",
		VALIDATION: {
			NAME_EXISTS: "A profile with this name already exists",
		},
	},
	PAGE_WELCOME: {
		HAS_EXPORTED_PROFILES: "Exported a Profile before?",
		IMPORT_PROFILE: "Import it here.",
		IMPORT_PROFILE_TITLE: "Import Profile",
		WITHOUT_PROFILES: {
			DESCRIPTION: "Create a new Profile to get started.",
			TITLE: "Create Profile",
		},
		WITH_PROFILES: {
			DESCRIPTION: "Choose from an existing Profile below or create a new Profile to get started.",
			TITLE: "Select Profile",
		},
	},
};
