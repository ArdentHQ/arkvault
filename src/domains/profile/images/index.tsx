import { ReactComponent as WelcomeModalBannerDarkBlue } from "./welcome-modal-banner-dark-blue.svg";
import { ReactComponent as WelcomeModalBannerDarkGreen } from "./welcome-modal-banner-dark-green.svg";
import { ReactComponent as WelcomeModalBannerLightBlue } from "./welcome-modal-banner-light-blue.svg";
import { ReactComponent as WelcomeModalBannerLightGreen } from "./welcome-modal-banner-light-green.svg";
import WelcomeModalStep1DarkBlue from "./welcome-modal-step-1-banner-dark-blue.png";
import WelcomeModalStep1DarkGreen from "./welcome-modal-step-1-banner-dark-green.png";
import WelcomeModalStep1LightBlue from "./welcome-modal-step-1-banner-light-blue.png";
import WelcomeModalStep1LightGreen from "./welcome-modal-step-1-banner-light-green.png";
import WelcomeModalStep2DarkBlue from "./welcome-modal-step-2-banner-dark-blue.png";
import WelcomeModalStep2DarkGreen from "./welcome-modal-step-2-banner-dark-green.png";
import WelcomeModalStep2LightBlue from "./welcome-modal-step-2-banner-light-blue.png";
import WelcomeModalStep2LightGreen from "./welcome-modal-step-2-banner-light-green.png";
import WelcomeModalStep3DarkBlue from "./welcome-modal-step-3-banner-dark-blue.png";
import WelcomeModalStep3DarkGreen from "./welcome-modal-step-3-banner-dark-green.png";
import WelcomeModalStep3LightBlue from "./welcome-modal-step-3-banner-light-blue.png";
import WelcomeModalStep3LightGreen from "./welcome-modal-step-3-banner-light-green.png";
import WelcomeModalStep4DarkBlue from "./welcome-modal-step-4-banner-dark-blue.png";
import WelcomeModalStep4DarkGreen from "./welcome-modal-step-4-banner-dark-green.png";
import WelcomeModalStep4LightBlue from "./welcome-modal-step-4-banner-light-blue.png";
import WelcomeModalStep4LightGreen from "./welcome-modal-step-4-banner-light-green.png";

export const profile = {
	WelcomeModalBannerDarkBlue,
	WelcomeModalBannerDarkGreen,
	WelcomeModalBannerLightBlue,
	WelcomeModalBannerLightGreen,
	WelcomeModalStep1DarkBlue,
	WelcomeModalStep1DarkGreen,
	WelcomeModalStep1LightBlue,
	WelcomeModalStep1LightGreen,
	WelcomeModalStep2DarkBlue,
	WelcomeModalStep2DarkGreen,
	WelcomeModalStep2LightBlue,
	WelcomeModalStep2LightGreen,
	WelcomeModalStep3DarkBlue,
	WelcomeModalStep3DarkGreen,
	WelcomeModalStep3LightBlue,
	WelcomeModalStep3LightGreen,
	WelcomeModalStep4DarkBlue,
	WelcomeModalStep4DarkGreen,
	WelcomeModalStep4LightBlue,
	WelcomeModalStep4LightGreen,
};
