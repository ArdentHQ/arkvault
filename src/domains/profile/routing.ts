import { ProfilePaths } from "@/router/paths";
import { RouteItem } from "@/router/router.types";
import preloadLazy from "@/utils/preload-lazy";

const CreateProfile = preloadLazy(() => import("./pages/CreateProfile"));
const ImportProfile = preloadLazy(() => import("./pages/ImportProfile"));
const Welcome = preloadLazy(() => import("./pages/Welcome"));

export const ProfileRoutes: RouteItem[] = [
	{
		component: CreateProfile,
		exact: true,
		path: ProfilePaths.CreateProfile,
	},
	{
		component: ImportProfile,
		exact: true,
		path: ProfilePaths.ImportProfile,
	},
	{
		component: Welcome,
		exact: true,
		path: ProfilePaths.Welcome,
	},
];
