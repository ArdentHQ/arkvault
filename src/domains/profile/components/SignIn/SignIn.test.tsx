/* eslint-disable @typescript-eslint/require-await */
import { Contracts } from "@ardenthq/sdk-profiles";
import userEvent from "@testing-library/user-event";
import React from "react";

import { SignIn } from "./SignIn";
import { translations } from "@/domains/profile/i18n";
import {
	act,
	env,
	getDefaultPassword,
	getPasswordProtectedProfileId,
	render,
	screen,
	waitFor,
} from "@/utils/testing-library";

let profile: Contracts.IProfile;

const submitID = "SignIn__submit-button";
const passwordInput = "SignIn__input--password";

describe("SignIn", () => {
	beforeEach(async () => {
		profile = env.profiles().findById(getPasswordProtectedProfileId());
		await env.profiles().restore(profile, getDefaultPassword());
	});

	beforeEach(() => {
		vi.useFakeTimers({ shouldAdvanceTime: true });
	});

	afterAll(() => {
		vi.useRealTimers();
	});

	it("should not render if not open", () => {
		const { asFragment } = render(<SignIn profile={profile} isOpen={false} onSuccess={vi.fn} />);

		expect(screen.queryByTestId("Modal__inner")).not.toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a modal", async () => {
		const { asFragment } = render(<SignIn isOpen={true} profile={profile} onSuccess={vi.fn} />);

		await waitFor(() => {
			expect(screen.getByTestId("Modal__inner")).toHaveTextContent(translations.MODAL_SIGN_IN.TITLE);
		});

		expect(screen.getByTestId("Modal__inner")).toHaveTextContent(translations.MODAL_SIGN_IN.DESCRIPTION);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should cancel sign in", async () => {
		const onCancel = vi.fn();

		render(<SignIn isOpen={true} profile={profile} onCancel={onCancel} onSuccess={vi.fn} />);

		await userEvent.click(screen.getByTestId("SignIn__cancel-button"));

		await waitFor(() => {
			expect(onCancel).toHaveBeenCalledWith(expect.objectContaining({ nativeEvent: expect.any(MouseEvent) }));
		});
	});

	it("should call onSuccess callback", async () => {
		const onSuccess = vi.fn();

		render(<SignIn isOpen={true} profile={profile} onSuccess={onSuccess} />);

		await userEvent.type(screen.getByTestId(passwordInput), getDefaultPassword());

		// wait for formState.isValid to be updated
		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		await userEvent.click(screen.getByTestId(submitID));

		await waitFor(() => {
			expect(onSuccess).toHaveBeenCalledWith(getDefaultPassword());
		});
	});

	it("should set an error if the password is invalid", async () => {
		const onSuccess = vi.fn();

		render(<SignIn isOpen={true} profile={profile} onSuccess={onSuccess} />);

		await userEvent.type(screen.getByTestId(passwordInput), "wrong password");

		// wait for formState.isValid to be updated
		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		await userEvent.click(screen.getByTestId(submitID));

		// wait for formState.isValid to be updated
		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		expect(screen.getByTestId("Input__error")).toBeVisible();
		expect(screen.getByTestId(submitID)).toBeDisabled();
	});

	it("should set an error if the password is invalid and count retries", async () => {
		const onSuccess = vi.fn();

		render(<SignIn isOpen={true} profile={profile} onSuccess={onSuccess} />);

		await userEvent.type(screen.getByTestId(passwordInput), "wrong password");

		// wait for formState.isValid to be updated
		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		await userEvent.click(screen.getByTestId(submitID));
		vi.advanceTimersByTime(20_000);

		await userEvent.type(screen.getByTestId(passwordInput), "wrong password");

		// wait for formState.isValid to be updated
		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		await userEvent.click(screen.getByTestId(submitID));
		vi.advanceTimersByTime(60_000);

		// wait for formState.isValid to be updated
		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		expect(screen.getByTestId("Input__error")).toBeVisible();
		expect(screen.getByTestId(submitID)).toBeDisabled();
	});

	it("should set an error and disable the input if the password is invalid multiple times", async () => {
		const onSuccess = vi.fn();

		render(<SignIn isOpen={true} profile={profile} onSuccess={onSuccess} />);

		for (const index of [1, 2, 3]) {
			await userEvent.type(screen.getByTestId(passwordInput), `wrong password ${index}`);

			// wait for form to be updated
			await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

			await userEvent.click(screen.getByTestId(submitID));

			// wait for form to be updated
			await expect(screen.findByTestId(submitID)).resolves.toBeVisible();
		}

		expect(screen.getByTestId(submitID)).toBeDisabled();
		expect(screen.getByTestId(passwordInput)).toBeDisabled();

		act(() => {
			vi.clearAllTimers();
		});

		// wait for form to be updated
		await expect(screen.findByTestId(submitID)).resolves.toBeVisible();

		await waitFor(() =>
			expect(screen.getByTestId("Input__error")).toHaveAttribute(
				"data-errortext",
				expect.stringMatching(/Maximum sign in attempts/),
			),
		);
	});
});
