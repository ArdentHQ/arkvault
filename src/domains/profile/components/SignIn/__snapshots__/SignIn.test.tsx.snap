// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignIn > should not render if not open 1`] = `<DocumentFragment />`;

exports[`SignIn > should render a modal 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-hidden bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-1jzz0qm"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pb-24 pt-10 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute right-0 top-0 z-10 mr-4 mt-4 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400 md:mr-8 md:mt-8"
        >
          <button
            class="h-8 w-8 css-1hei9vx"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg
                  id="cross"
                  viewBox="0 0 20 20"
                  x="0"
                  xml:space="preserve"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0"
                >
                  <g
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  >
                    <path
                      d="M.954 18.981l18-18M18.954 18.981l-18-18"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative flex flex-1 flex-col space-y-1.5"
        >
          <h2
            class="mb-0 mt-1 text-lg font-bold md:pt-0 md:text-2xl"
          >
            Sign In
          </h2>
          <div
            class="flex flex-1 flex-col"
          >
            <div
              class="whitespace-pre-line pr-10 text-sm text-theme-secondary-text md:pr-0 md:text-base"
            >
              Sign in to access your Profile.
            </div>
            <div
              class="mt-8"
            >
              <div
                class="flex items-center space-x-3"
              >
                <div
                  class="shrink-0 css-1px82g5"
                  data-testid="Avatar"
                >
                  <div
                    class="inline-flex h-full w-full items-center justify-center overflow-hidden align-middle rounded"
                  >
                    <img
                      alt="Jane Doe"
                      data-testid="ProfileAvatar__svg"
                      src="data:image/svg+xml;utf8,<svg version="1.1" xmlns="http://www.w3.org/2000/svg" class="picasso" width="100" height="100" viewBox="0 0 100 100"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill="rgb(0, 150, 136)" width="100" height="100"/><circle r="55" cx="20" cy="60" fill="rgb(233, 30, 99)"/><circle r="60" cx="50" cy="70" fill="rgb(255, 193, 7)"/><circle r="45" cx="30" cy="30" fill="rgb(63, 81, 181)"/></svg>"
                      title="Jane Doe"
                    />
                    <span
                      class="absolute font-semibold text-white"
                    >
                      JA
                    </span>
                  </div>
                </div>
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-sm font-semibold text-theme-secondary-500 dark:text-theme-secondary-700"
                  >
                    Name
                  </p>
                  <p
                    class="font-semibold text-theme-text"
                    data-testid="profile-card__user--name"
                  >
                    <span>
                      Jane Doe
                    </span>
                  </p>
                </div>
              </div>
            </div>
            <div
              class="border-theme-secondary-300 dark:border-theme-secondary-800 css-ppmbkz"
              type="horizontal"
            />
            <form
              data-testid="Form"
            >
              <fieldset
                class="flex min-w-0 flex-col css-1bcidjh"
              >
                <label
                  class="FormLabel mb-2 flex text-sm font-semibold leading-[17px] text-theme-secondary-text transition-colors duration-100"
                  data-testid="FormLabel"
                  for="password"
                >
                  Password
                </label>
                <div
                  class="css-n1b0kl"
                >
                  <div
                    class="relative flex h-full flex-1"
                  >
                    <input
                      autocomplete="off"
                      class="no-ligatures w-full border-none css-1vs1awz"
                      data-testid="SignIn__input--password"
                      name="password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                    data-testid="Input__addon-end"
                  >
                    <div
                      class=""
                    >
                      <button
                        class="ring-focus relative flex h-full w-full items-center justify-center text-2xl focus:outline-none"
                        data-ring-focus-margin="-m-1"
                        data-testid="InputPassword__toggle"
                        type="button"
                      >
                        <div
                          class="css-1i4b0eq"
                          height="20"
                          width="20"
                        >
                          <svg
                            id="eye"
                            viewBox="0 0 20 20"
                            x="0"
                            xml:space="preserve"
                            xmlns="http://www.w3.org/2000/svg"
                            y="0"
                          >
                            <g
                              fill="none"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            >
                              <path
                                d="M10 4.6c-3.2-.1-6.5 2.2-8.6 4.5-.5.5-.5 1.3 0 1.8 2 2.3 5.4 4.6 8.6 4.5 3.3-.1 6.5-2.3 8.6-4.5.5-.5.5-1.3 0-1.8-2.1-2.3-5.4-4.6-8.6-4.5z"
                              />
                              <path
                                d="M13 10c0 1.7-1.3 3-3 3s-3-1.3-3-3 1.3-3 3-3c1.2 0 3 .7 3 3z"
                              />
                            </g>
                          </svg>
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </fieldset>
              <div
                class="css-j6ajdm"
              >
                <button
                  class="css-6419sw"
                  data-testid="SignIn__cancel-button"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    Cancel
                  </div>
                </button>
                <button
                  class="css-fl4fi0"
                  data-testid="SignIn__submit-button"
                  type="submit"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    Sign In
                  </div>
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
